README.md

```diff
@@ -1,22 +1,134 @@
+   - In your terminal, navigate to the project directory and run the following command to start the application:
+     ```bash
+     make run
+     ```
+   - Leave this terminal session running and wait until you see that [Nest local](http://localhost:8000/api/v1) is responding.
+
+1. **Load Initial Data**:
+   - Open a new terminal session and run the following command to populate the database with initial data from fixtures:
+     ```bash
+     make load-data
+     ```
+
+1. **Index Data**:
+   - In the same terminal session, run the following command to index the data:
+     ```bash
+     make index-data
+     ```
+
+1. **Verify API Endpoints**:
+   - Check that the data is available via these API endpoints:
+     - [Projects Endpoint](http://localhost:8000/api/v1/owasp/search/project)
+     - [Issues Endpoint](http://localhost:8000/api/v1/owasp/search/issue)
+
+### Optional Steps (for managing or fetching data):
+
+If you plan to manage or fetch data, follow these additional steps:
+
+1. **Create a Super User**:
+   - Run the following command to create a super user for accessing the admin interface:
+     ```bash
+     make setup
+     ```
+
+1. **Generate a GitHub Personal Access Token**:
+   - Create a GitHub [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+1. **Update Environment Variables with GitHub Token**:
+   - Open `backend/.env/local` again and update it with your GitHub token:
+     ```plaintext
+     GITHUB_TOKEN=<your_github_token>
+     ```
+
+1. **Sync Local Database Data**:
+   - Now you should be able to run the following command to sync your local database data with GitHub:
+     ```bash
+     make sync
+     ```
+
+
+## Linting and Formatting Setup
+
+To ensure code quality, we utilize ESLint and Prettier with Husky for pre-commit hooks.
+
+### Installing Dependencies
+
+Make sure you have the necessary dependencies installed:
+
+```bash
+npm install --save-dev eslint prettier husky lint-staged eslint-plugin-react eslint-plugin-react-hooks @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier eslint-plugin-prettier
+```
+### Configure Husky
+To set up Husky for pre-commit hooks, run:
+
+```bash
+npm run prepare
+```
+
+### Pre-commit Hook Script
+Ensure the `.husky/pre-commit` file consists of the following content:
+
+```bash
+#!/bin/sh
+# .husky/pre-commit
+echo "Running pre-commit hook..."
+echo "Running lint-staged..."
+export NODE_OPTIONS="--max-old-space-size=4096"
+npx lint-staged
+
+# Optional: Run ESLint and Prettier directly for debugging
+eslint . --fix --config eslint.config.js
+prettier --write .
+```
+
+
+### Linting and Formatting Commands
+You can manually run ESLint and Prettier to check your setup:
+   - To lint your code:
+```bash
+npx eslint . --fix --config eslint.config.js
+```
+
+  - To format your code with Prettier:
+```bash
+npx prettier --write .
```

Comment:
Ideally we want to have a single command, maybe another Makefile target if possible to use it as another step for `make pre-commit` command.

It should be easy for a developer to make the code to comply with the expected requirements/style by using just a single command.

Reply:
@arkid15r I put these commands so that user could manually check if their lint and prettier format setup was correct. Originally, running the `husky pre-commit` command will perform both of these tasks

Reply:
Let's think on how to simplify it after merging this PR